src/tools/rbd/action/Create.cc

```diff
@@ -25,7 +26,133 @@ void get_arguments(po::options_description *positional,
                    po::options_description *options) {
   at::add_image_spec_options(positional, options, at::ARGUMENT_MODIFIER_NONE);
   at::add_create_image_options(options, true);
+  options->add_options()
+    (at::IMAGE_THICK.c_str(), po::bool_switch(), "thick image provisioned");
   at::add_size_option(options);
+  at::add_no_progress_option(options);
+}
+
+int write_data(librbd::Image &image, bool no_progress)
+{
+  uint64_t concurr =
+    g_conf->get_val<int64_t>("rbd_concurrent_management_ops");
+
+  uint64_t image_size;
+  const int chunk_size = 1024 * 1024; // 1MB
+  const int block_size = 512; // 512 Bytes
+  char write_buff[block_size];
+  uint64_t chunk_num;
+  uint64_t complete_count = 0;
+  uint64_t target_chunk = 0;
+  uint64_t write_offset;
+  int max_aio = 0;
+  int r = 0;
+  ceph::bufferlist bl;
+  librbd::RBD::AioCompletion *cvec[concurr];
+  utils::ProgressContext pc("Writing data for thick provisioning", no_progress);
+
+  if (image.size(&image_size) != 0) {
+    r = -EINVAL;
+    goto err_size;
+  }
+
+  chunk_num = image_size / chunk_size;
+
+  // If write_buf has zero data, aio_writesame doesn't write
```

Comment:
I'd prefer to see a new "rbd_discard_on_zeroed_write_same" option added to "src/common/options.cc" and modify "src/librbd/librbd.cc" write same API methods to only issue discards if that option is enabled. That option should default to true but this method, before opening the image, would override the config option to set it to false.

Reply:
I cannot find the "rbd_discard_on_zeroed_write_same" option in "src/common/options.cc" and other files from the latest master branch. Is there the option in other branches?